plugins/nsmserver/nsmserver_client.go

```diff
@@ -66,7 +66,34 @@ func (n nsmClientEndpoints) RequestDiscovery(ctx context.Context, cr *nsmconnect
 }
 
 func (n *nsmClientEndpoints) RequestAdvertiseChannel(ctx context.Context, cr *nsmconnect.ChannelAdvertiseRequest) (*nsmconnect.ChannelAdvertiseResponse, error) {
-	return nil, status.Error(codes.InvalidArgument, "Not Implemented...")
+	n.logger.Printf("received Channel advertisement...")
+	for _, c := range cr.NetmeshChannel {
+		n.logger.Infof("For NetworkService: %s channel: %s channel's socket location: %s", c.NetworkServiceName, c.Metadata.Name, c.SocketLocation)
+		nsName := c.NetworkServiceName
+		nsNamespace := "default"
+		if c.Metadata.Namespace != "" {
+			nsNamespace = c.Metadata.Namespace
+		} else {
+			c.Metadata.Namespace = "default"
+		}
+
+		ns := n.objectStore.GetNetworkService(nsName, nsNamespace)
+		if ns != nil {
+			n.logger.Infof("Found existing NetworkService %s/%s in the Object Store, will add channel %s to its list of channels",
+				nsName, nsNamespace, c.Metadata.Name)
+			// Since it was discovered that NetworkService Object exists, calling method to add the channel to NetworkService.
+			if err := n.objectStore.AddChannelToNetworkService(nsName, nsNamespace, c); err != nil {
+				n.logger.Error("failed to add channel %s/%s to network service %s with error: %+v", nsNamespace, nsName, c.Metadata.Name, err)
+				return &nsmconnect.ChannelAdvertiseResponse{Success: false}, err
+			}
+			n.logger.Infof("Channel %s/%s has been successfully added to network service %s/%s in the Object Store",
+				c.Metadata.Namespace, c.Metadata.Name, nsName, nsNamespace)
+		} else {
+			n.logger.Infof("NetworkService %s/%s is not found in the Object Store", nsName, nsNamespace)
+			// TODO (sbezverk) need to decide, should the new NetworkService object be created, or it is failure scenario?
```

Comment:
Did you want to add the logic here to create the NetworkService? That would truly complete this PR.

Reply:
@mestery @edwarnicke @fkautz As discussed over IRC channel, I do not feel comfortable at least at this point to create NetworkService from NSM code as it could trigger huge number of NS being created for every misspelled network service name in the channel advertisement,  I for now it should be cluster admin responsibility to create NetworkService objects on needed basis and then NSEs will populate them with corresponding channels. WDYT?

Reply:
I'm fine with this being an error instead.

Reply:
What is the likeliness that we will see many misspelled network services?

Reply:
unintentional very little but just imagine an attack which can kill a cluster with thousands of these.